### ISSUE TITLE
lazy json parsing affects dict semantics

### ISSUE THREAD

**Issue by vitillo** [mozilla/python_moztelemetry#8]
```
foo = _lazyjson('{"foo": 10}')
print foo
print dict(foo)
```

returns 

```
{u'foo': 10}
{}
```

**Comment by vitillo** [comment]
There reason is due to the [way](https://github.com/python/cpython/blob/master/Objects/dictobject.c#L2041) python creates a new dict from an existing one.

In the above example, as the internal wrapper subclasses *dict*, the constructor uses the wrapper’s keys directly instead of doing the more expensive thing which is to invoke PyMapping_Keys() and PyObject_GetItem(). As the wrapper has no keys though the new dict is empty.

Not subclassing from *dict* might be an option but that would in turn cause interoperability issues with external libraries that expect a native type instead of something that looks like a native type...

Since the issue is not triggered with objects returned from a Spark action, likely due to cloudpickle, I am inclined to leave this open for now.

**Comment by mreid-moz** [comment]
Fixed by #160
